### ISSUE TITLE
libvpx fails to cross-compile to armv7l-hf-multiplatform

### ISSUE THREAD

**Issue by marmay** [NixOS/nixpkgs#208746]
### Steps To Reproduce
1. Make sure that the system is set up for cross-compiling (extra-platforms).
2. `nix-build '<nixpkgs>' -A pkgsCross.armv7l-hf-multiplatform.libvpx`

### Build log
```
Configuring for target 'armv7l-linux-gcc'
  enabling armv7l
# ...
    [LD] libvpx.so.7.1.0
    [AR] libvpx_g.a
    [AR] libvpxrc_g.a
    [STRIP] libvpxrc.a < libvpxrc_g.a
    [STRIP] libvpx.a < libvpx_g.a
/nix/store/y7bs7f9j931b2hps70dc1d165m45cpib-armv7l-unknown-linux-gnueabihf-binutils-2.39/bin/armv7l-unknown-linux-gnueabihf-ld: vpx_dsp/vpx_dsp_rtcd.c.o: in function `setup_rtcd_internal':
vpx_dsp_rtcd.c:(.text+0x1014): undefined reference to `vpx_avg_4x4_neon'
# ...
```

### Additional context
The following patch fixes the cross-build I am looking for (but it might break other builds, I don't know):
```
diff --git a/pkgs/development/libraries/libvpx/default.nix b/pkgs/development/libraries/libvpx/default.nix
index d78a0885..999ef381 100644
--- a/pkgs/development/libraries/libvpx/default.nix
+++ b/pkgs/development/libraries/libvpx/default.nix
@@ -148,7 +148,7 @@ stdenv.mkDerivation rec {
     # libvpx darwin targets include darwin version (ie. ARCH-darwinXX-gcc, XX being the darwin version)
     # See all_platforms: https://github.com/webmproject/libvpx/blob/master/configure
     # Darwin versions: 10.4=8, 10.5=9, 10.6=10, 10.7=11, 10.8=12, 10.9=13, 10.10=14
-    "--force-target=${stdenv.hostPlatform.parsed.cpu.name}-${stdenv.hostPlatform.parsed.kernel.name}${
+    "--target=${stdenv.hostPlatform.parsed.cpu.arch}-${stdenv.hostPlatform.parsed.kernel.name}${
             if stdenv.hostPlatform.isDarwin then
               if      stdenv.hostPlatform.osxMinVersion == "10.10" then "14"
               else if stdenv.hostPlatform.osxMinVersion == "10.9"  then "13"
```
Issue seems to be that the configure script of libvpx does not know the target (as a result, we need to force it), which results in a mess. By using `arch` instead of `name`, the target is understood and everything works for me.

### Notify maintainers
@codyopel 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.85, NixOS, 22.11 (Raccoon), 22.11.1198.913a47cd064`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(markus): `""`
 - channels(root): `"home-manager-22.11.tar.gz, nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by RossComputerGuy** [comment]
I'm experiencing this issue as well, I'll test your patch in ExpidusOS and I'll reply back with whether or not it works.

**Comment by RossComputerGuy** [comment]
```
  enabling webm_io
  enabling libyuv
  disabling decode_perf_tests
  disabling encode_perf_tests
  disabling multi_res_encoding
  enabling vp9_temporal_denoising
  disabling coefficient_range_checking
  disabling vp9_highbitdepth
  disabling experimental
Unrecognized toolchain 'armv6l-linux-gcc'

Configuration failed. This could reflect a misconfiguration of your
toolchains, improper options selected, or another problem. If you
don't see any useful error messages above, the next step is to look
at the configure error log file (config.log) to determine what
configure was trying to do when it died.
```
This is gonna be a bit to fix up

**Comment by RossComputerGuy** [comment]
```
  disabling multi_res_encoding
  enabling vp9_temporal_denoising
  disabling coefficient_range_checking
  disabling vp9_highbitdepth
  disabling experimental
Unrecognized toolchain 'armv6-linux-gcc'

Configuration failed. This could reflect a misconfiguration of your
toolchains, improper options selected, or another problem. If you
don't see any useful error messages above, the next step is to look
at the configure error log file (config.log) to determine what
configure was trying to do when it died
```
After applying your patch, I wasn't able to build it. I'm going to look into what `libvpx` wants it to be so it could compile.

**Comment by RossComputerGuy** [comment]
I was able to successfully cross compile `libvpx`, I'll work on bundling [this](https://github.com/ExpidusOS/sdk/commit/1f488bb370a4e0afcdff9f1e528c66fdc3ac494c) into a PR later.

**Comment by RossComputerGuy** [comment]
@uninsane Done. Fixed by #317477.
